docs/src/main/asciidoc/_elytron/Keycloak_SAML_Integration.adoc

```diff
@@ -0,0 +1,191 @@
+[[Keycloak_SAML_Integration]]
+= Using Keycloak SAML
+
+To secure applications deployed to WildFly using Keycloak SAML, a Galleon feature pack provided
+by the Keycloak project needs to be used. This feature pack automatically installs the Keycloak
+SAML adapter that results in the `keycloak-saml` subsystem being added to WildFly.
+
+There are a few ways to install the Keycloak Galleon Feature Pack that provides the SAML adapter.
+
+[[galleon-cli]]
+== Galleon CLI
+
+A server installation can be provisioned using the Galleon CLI.
+
+More information can be found in the <<Galleon_Guide.adoc#Galleon_Guide,Provisioning WildFly with Galleon>>
+documentation, but assuming you have provisioned a WildFly installation containing the `web-server` layer
+with a command similar to the following:
+
+[source]
+----
+galleon.sh install wildfly:current \
+    --layers=web-server --dir=wildfly
+----
+
+The Keycloak SAML adapter can then be added to the WildFly installation with the following command:
+
+[source]
+----
+TODO: Update the version here when we know what it is
+galleon.sh install org.keycloak:keycloak-saml-adapter-galleon-pack:999 --layers=keycloak-client-saml --dir=wildfly
+----
+
+[NOTE]
+====
+Unlike the WildFly feature pack, this Keycloak feature pack is not part of a universe and so a fully
+qualified `group:artifact:version` reference to the feature pack is required.
+====
+
+From this point, applications can be configured as described in the
+https://www.keycloak.org/docs/latest/securing_apps/index.html#_jboss_adapter[Keycloak documentation].
+
+[[bootable-jar]]
+== Bootable JAR
+
+The next installation option is if you are creating a bootable JAR for your deployment and want to
+add the Keycloak SAML adapter to secure the deployment.  More details about bootable JAR support can
+be found in the <<Bootable_Guide.adoc#wildfly_bootable_JAR_development,Bootable JAR Guide>>.
+
+The following is an example plug-in configuration to create a bootable JAR for a web application
+using both the `web-server` and `keycloak-client-saml` layers:
+
+[source,xml,options="nowrap"]
+----
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-jar-maven-plugin</artifactId>
+    <version>${version.wildfly.jar.maven.plugin}</version>
+    <configuration>
+        <feature-packs>
+            <feature-pack>
+                <location>wildfly@maven(org.jboss.universe:community-universe):current</location>
+            </feature-pack>
+            <feature-pack>
+                <groupId>org.keycloak</groupId>
+                <artifactId>keycloak-adapter-galleon-pack</artifactId>
```

Comment:
@fjuma should this be `<artifactId>keycloak-saml-adapter-galleon-pack</artifactId>`?